skrub/_datetime_encoder.py

```diff
@@ -284,16 +346,57 @@ def fit_transform(self, column, y=None):
                 f"Column {sbd.name(column)!r} does not have Date or Datetime dtype."
             )
         if self.resolution is None:
-            self.extracted_features_ = []
+            self._partial_features = []
         else:
             idx_level = _TIME_LEVELS.index(self.resolution)
             if _is_date(column):
                 idx_level = min(idx_level, _TIME_LEVELS.index("day"))
-            self.extracted_features_ = _TIME_LEVELS[: idx_level + 1]
+            self._partial_features = _TIME_LEVELS[: idx_level + 1]
         if self.add_total_seconds:
-            self.extracted_features_.append("total_seconds")
+            self._partial_features.append("total_seconds")
         if self.add_weekday:
-            self.extracted_features_.append("weekday")
+            self._partial_features.append("weekday")
+        if self.add_day_of_year:
+            self._partial_features.append("day_of_year")
+
+        self.extracted_features_ = [f"{feat}" for feat in self._partial_features]
+
+        col_name = sbd.name(column)
+
+        # Adding transformers for periodic encoding
+        self._required_transformers = {}
```

Comment:
could we have a name more specific than "required", for example `self._periodic_encoders`?